injected/src/features/favicon.js

```diff
@@ -51,18 +53,35 @@ export default Favicon;
 
 /**
  * @param {()=>void} changeObservedCallback
- * @param {Element} [target]
  */
-function monitor(changeObservedCallback, target = document.head) {
+function monitor(changeObservedCallback) {
+    const target = document.head;
+    if (!target) return;
+
     const observer = new MutationObserver((mutations) => {
         for (const mutation of mutations) {
             if (mutation.type === 'attributes' && mutation.target instanceof HTMLLinkElement) {
                 changeObservedCallback();
                 break;
             }
+            if (mutation.type === 'childList') {
+                for (const addedNode of mutation.addedNodes) {
+                    if (addedNode instanceof HTMLLinkElement) {
+                        changeObservedCallback();
+                        break;
```

Comment:
I think you meant to break from the outer loop too?

Reply:
Oh yeah, you're correct! thanks

Reply:
I'll just return instead